### PR TITLE
r/aws_cloudfront_realtime_log_config: New resource

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes: https://github.com/terraform-providers/terraform-provider-aws/issues/14944.

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudFrontRealtimeLogConfig_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudFrontRealtimeLogConfig_ -timeout 120m
=== RUN   TestAccAWSCloudFrontRealtimeLogConfig_basic
=== PAUSE TestAccAWSCloudFrontRealtimeLogConfig_basic
=== RUN   TestAccAWSCloudFrontRealtimeLogConfig_disappears
=== PAUSE TestAccAWSCloudFrontRealtimeLogConfig_disappears
=== RUN   TestAccAWSCloudFrontRealtimeLogConfig_updates
=== PAUSE TestAccAWSCloudFrontRealtimeLogConfig_updates
=== CONT  TestAccAWSCloudFrontRealtimeLogConfig_basic
=== CONT  TestAccAWSCloudFrontRealtimeLogConfig_updates
=== CONT  TestAccAWSCloudFrontRealtimeLogConfig_disappears
    resource_aws_cloudfront_realtime_log_config_test.go:118: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_disappears (62.69s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_basic (64.35s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_updates (110.45s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	110.498s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners\|TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners\|TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames -timeout 120m
=== RUN   TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners
=== PAUSE TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners
=== RUN   TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn
=== PAUSE TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn
=== RUN   TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn
=== PAUSE TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn
=== RUN   TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames
=== PAUSE TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames
=== CONT  TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners
=== CONT  TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames
=== CONT  TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn
=== CONT  TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners (205.48s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames (206.56s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn (281.89s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn (281.95s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	281.989s
```

### CODE REVIEW

**Review by DrFaust92 — APPROVED** [review]
LGTM 

```
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_ForwardedValues_Headers (231.60s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners (235.72s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn (583.97s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_basic (96.68s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_disappears (87.32s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_updates (170.19s)
```

**Comment by ken5scal** [comment]
Can some one review/merge this?

**Comment by richerve** [comment]
what else is required to merge this? We also depend on this to debug request headers, and that information is not provided by regular logs.

**Comment by acsgh** [comment]
Hello @ewbankkit, we can also need this feature for our ongoings developments. Is there a reason why this PR is not merged?

Many thank!

**Comment by CajuCLC** [comment]
Hi folks. Is there an ETA for this PR to be merged?
Thanks.

**Comment by spider-network** [comment]
We are also waiting for this merge.

**Comment by seanturner026** [comment]
Hello @ewbankkit, we also need this feature for our ongoings developments. Is there a reason why this PR is not merged?

Many thanks!

**Comment by rajholla** [comment]
Hello @ewbankkit, 
We are also looking forward to use this feature.

**Comment by ewbankkit** [comment]
Rebase to fix merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudFrontRealtimeLogConfig_\|TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners\|TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudFrontRealtimeLogConfig_\|TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn\|TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners\|TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames -timeout 120m
=== RUN   TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners
=== PAUSE TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners
=== RUN   TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn
=== PAUSE TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn
=== RUN   TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn
=== PAUSE TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn
=== RUN   TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames
=== PAUSE TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames
=== RUN   TestAccAWSCloudFrontRealtimeLogConfig_basic
=== PAUSE TestAccAWSCloudFrontRealtimeLogConfig_basic
=== RUN   TestAccAWSCloudFrontRealtimeLogConfig_disappears
=== PAUSE TestAccAWSCloudFrontRealtimeLogConfig_disappears
=== RUN   TestAccAWSCloudFrontRealtimeLogConfig_updates
=== PAUSE TestAccAWSCloudFrontRealtimeLogConfig_updates
=== CONT  TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners
=== CONT  TestAccAWSCloudFrontRealtimeLogConfig_updates
=== CONT  TestAccAWSCloudFrontRealtimeLogConfig_disappears
=== CONT  TestAccAWSCloudFrontRealtimeLogConfig_basic
=== CONT  TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames
=== CONT  TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn
=== CONT  TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_disappears (62.91s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_basic (64.22s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_updates (107.78s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames (182.38s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners (184.36s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn (212.25s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn (212.83s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	212.944s
```

**Comment by srgustafson8** [comment]
hi - any update on this?

File: website/docs/r/cloudfront_realtime_log_config.html.markdown [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+* `id` - The ID of the CloudFront real-time log configuration.
+* `arn` - The ARN (Amazon Resource Name) of the CloudFront real-time log configuration.
+
+## Import
+
+CloudFront real-time log configurations can be imported using the ARN, e.g.
+
+```
+$ terraform import aws_cloudfront_realtime_log_config.example TODO
```
Comment (anGie44) [inline website/docs/r/cloudfront_realtime_log_config.html.markdown:108]: Small Nit: 
```suggestion
$ terraform import aws_cloudfront_realtime_log_config.example arn:aws:cloudfront::111122223333:realtime-log-config/ExampleNameForRealtimeLogConfig
```

File: aws/resource_aws_cloudfront_realtime_log_config.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+func resourceAwsCloudFrontRealtimeLogConfigDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	log.Printf("[DEBUG] Deleting CloudFront Real-time Log Config (%s)", d.Id())
+	_, err := conn.DeleteRealtimeLogConfig(&cloudfront.DeleteRealtimeLogConfigInput{
+		ARN: aws.String(d.Id()),
+	})
+
+	if isAWSErr(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig, "") {
```
Comment (anGie44) [inline aws/resource_aws_cloudfront_realtime_log_config.go:173]: Nit:
```suggestion
	if tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig) {
```

Comment (ewbankkit) [reply aws/resource_aws_cloudfront_realtime_log_config.go:173]: I'll add the `tfawserr` import and fix this (and any others).

File: aws/resource_aws_cloudfront_realtime_log_config.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+	return resourceAwsCloudFrontRealtimeLogConfigRead(d, meta)
+}
+
+func resourceAwsCloudFrontRealtimeLogConfigRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	logConfig, err := finder.RealtimeLogConfigByARN(conn, d.Id())
+
+	if isAWSErr(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig, "") {
```
Comment (anGie44) [inline aws/resource_aws_cloudfront_realtime_log_config.go:116]: tech-debt nit:
```suggestion
	if !d.IsNewResource && tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig) {
```

File: aws/resource_aws_cloudfront_realtime_log_config.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+	if err != nil {
+		return fmt.Errorf("error reading CloudFront Real-time Log Config (%s): %w", d.Id(), err)
+	}
+
+	if logConfig == nil {
+		log.Printf("[WARN] CloudFront Real-time Log Config (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
```
Comment (anGie44) [inline aws/resource_aws_cloudfront_realtime_log_config.go:130]:
```suggestion
	if logConfig == nil {
	        if d.IsNewResource() {
	            return fmt.Errorf("error reading CloudFront Real-time Log Config (%s): not found", d.Id())
	        }
		log.Printf("[WARN] CloudFront Real-time Log Config (%s) not found, removing from state", d.Id())
		d.SetId("")
		return nil
	}
```

File: aws/resource_aws_cloudfront_realtime_log_config.go (hunk 3 above)
Comment (ewbankkit) [reply aws/resource_aws_cloudfront_realtime_log_config.go:116]: Arriving soon in a new commit.

File: aws/resource_aws_cloudfront_realtime_log_config.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+		Fields:       expandStringSet(d.Get("fields").(*schema.Set)),
+		Name:         aws.String(d.Get("name").(string)),
+		SamplingRate: aws.Int64(int64(d.Get("sampling_rate").(int))),
+	}
+
+	log.Printf("[DEBUG] Creating CloudFront Real-time Log Config: %s", input)
+	output, err := conn.CreateRealtimeLogConfig(input)
+
+	if err != nil {
+		return fmt.Errorf("error creating CloudFront Real-time Log Config: %w", err)
```
Comment (anGie44) [inline aws/resource_aws_cloudfront_realtime_log_config.go:103]: nit: for more context, return the name. might be easier to store the name initially in var up top

```suggestion
		return fmt.Errorf("error creating CloudFront Real-time Log Config (%s): %w", d.Get("name").(string), err)
```

**Review by anGie44 — APPROVED** [review]
Just a couple nitpicks, otherwise LGTM 👍 

Output of acceptance tests (commercial):
```
--- PASS: TestCloudFrontStructure_expandCachedMethods (0.00s)
--- PASS: TestCloudFrontStructure_expandCloudFrontDefaultCacheBehavior (0.00s)
--- PASS: TestCloudFrontStructure_expandLambdaFunctionAssociations (0.00s)
--- PASS: TestCloudFrontStructure_expandHeaders (0.00s)
--- PASS: TestCloudFrontStructure_flattenTrustedSigners (0.00s)
--- PASS: TestCloudFrontStructure_expandlambdaFunctionAssociations_empty (0.00s)
--- PASS: TestCloudFrontStructure_expandForwardedValues (0.00s)
--- PASS: TestCloudFrontStructure_expandTrustedSigners (0.00s)
--- PASS: TestCloudFrontStructure_expandQueryStringCacheKeys (0.00s)
--- PASS: TestCloudFrontStructure_expandAllowedMethods (0.00s)
--- PASS: TestCloudFrontStructure_flattenlambdaFunctionAssociations (0.00s)
--- PASS: TestCloudFrontStructure_expandTrustedSigners_empty (0.00s)
--- PASS: TestCloudFrontStructure_flattenForwardedValues (0.00s)
--- PASS: TestCloudFrontStructure_flattenAllowedMethods (0.00s)
--- PASS: TestCloudFrontStructure_flattenCookieNames (0.00s)
--- PASS: TestCloudFrontStructure_flattenCookiePreference (0.00s)
--- PASS: TestCloudFrontStructure_flattenCachedMethods (0.00s)
--- PASS: TestCloudFrontStructure_expandCookieNames (0.00s)
--- PASS: TestCloudFrontStructure_expandCookiePreference (0.00s)
--- PASS: TestCloudFrontStructure_flattenQueryStringCacheKeys (0.00s)
--- PASS: TestCloudFrontStructure_expandOriginGroups (0.00s)
--- PASS: TestCloudFrontStructure_flattenOrigins (0.00s)
--- PASS: TestCloudFrontStructure_flattenOriginGroups (0.00s)
--- PASS: TestCloudFrontStructure_expandOrigins (0.00s)
--- PASS: TestCloudFrontStructure_flattenHeaders (0.00s)
--- PASS: TestCloudFrontStructure_expandOrigin (0.03s)
--- PASS: TestCloudFrontStructure_expandCustomHeaders (0.00s)
--- PASS: TestCloudFrontStructure_expandOriginCustomHeader (0.00s)
--- PASS: TestCloudFrontStructure_flattenOrigin (0.00s)
--- PASS: TestCloudFrontStructure_flattenOriginCustomHeader (0.00s)
--- PASS: TestCloudFrontStructure_flattenCustomHeaders (0.00s)
--- PASS: TestCloudFrontStructure_expandCustomOriginConfig (0.00s)
--- PASS: TestCloudFrontStructure_flattenCustomOriginConfig (0.00s)
--- PASS: TestCloudFrontStructure_expandCustomOriginConfigSSL (0.00s)
--- PASS: TestCloudFrontStructure_expandCustomErrorResponse_emptyResponseCode (0.00s)
--- PASS: TestCloudFrontStructure_expandCustomErrorResponses (0.00s)
--- PASS: TestCloudFrontStructure_flattenCustomErrorResponses (0.00s)
--- PASS: TestCloudFrontStructure_expandS3OriginConfig (0.00s)
--- PASS: TestCloudFrontStructure_flattenCustomOriginConfigSSL (0.00s)
--- PASS: TestCloudFrontStructure_flattenS3OriginConfig (0.00s)
--- PASS: TestCloudFrontStructure_expandCustomErrorResponse (0.00s)
--- PASS: TestCloudFrontStructure_expandLoggingConfig (0.00s)
--- PASS: TestCloudFrontStructure_flattenAliases (0.00s)
--- PASS: TestCloudFrontStructure_expandLoggingConfig_nilValue (0.00s)
--- PASS: TestCloudFrontStructure_flattenCustomErrorResponse (0.00s)
--- PASS: TestCloudFrontStructure_expandGeoRestriction_whitelist (0.00s)
--- PASS: TestCloudFrontStructure_expandAliases (0.00s)
--- PASS: TestCloudFrontStructure_flattenGeoRestriction_whitelist (0.00s)
--- PASS: TestCloudFrontStructure_expandRestrictions (0.00s)
--- PASS: TestCloudFrontStructure_expandViewerCertificate_iam_certificate_id (0.00s)
--- PASS: TestCloudFrontStructure_expandViewerCertificate_acm_certificate_arn (0.00s)
--- PASS: TestCloudFrontStructure_expandGeoRestriction_no_items (0.00s)
--- PASS: TestCloudFrontStructure_flattenGeoRestriction_no_items (0.00s)
--- PASS: TestCloudFrontStructure_expandViewerCertificate_cloudfront_default_certificate (0.00s)
--- PASS: TestAccAWSCloudFrontDistribution_Origin_EmptyOriginID (11.59s)
--- PASS: TestAccAWSCloudFrontDistribution_Origin_EmptyDomainName (11.71s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_TrustedSigners (175.99s)
--- PASS: TestAccAWSCloudFrontDistribution_disappears (187.44s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_RealtimeLogConfigArn (222.92s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_ForwardedValues_Headers (241.77s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Cookies_WhitelistedNames (230.25s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_ForwardedValues_Headers (236.29s)
--- PASS: TestAccAWSCloudFrontDistribution_OrderedCacheBehavior_RealtimeLogConfigArn (252.45s)
--- PASS: TestAccAWSCloudFrontDistribution_ViewerCertificate_AcmCertificateArn (249.87s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_disappears (55.94s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_basic (58.26s)
--- PASS: TestAccAWSCloudFrontRealtimeLogConfig_updates (100.08s)
--- PASS: TestAccAWSCloudFrontDistribution_DefaultCacheBehavior_ForwardedValues_Cookies_WhitelistedNames (371.88s)
--- PASS: TestAccAWSCloudFrontDistribution_HTTP11Config (384.25s)
--- PASS: TestAccAWSCloudFrontDistribution_noOptionalItemsConfig (386.58s)
--- PASS: TestAccAWSCloudFrontDistribution_customOrigin (387.70s)
--- PASS: TestAccAWSCloudFrontDistribution_S3Origin (388.00s)
--- PASS: TestAccAWSCloudFrontDistribution_IsIPV6EnabledConfig (389.61s)
--- PASS: TestAccAWSCloudFrontDistribution_noCustomErrorResponseConfig (389.96s)
--- PASS: TestAccAWSCloudFrontDistribution_RetainOnDelete (390.05s)
--- PASS: TestAccAWSCloudFrontDistribution_multiOrigin (391.51s)
--- PASS: TestAccAWSCloudFrontDistribution_orderedCacheBehavior (392.27s)
--- PASS: TestAccAWSCloudFrontDistribution_ViewerCertificate_AcmCertificateArn_ConflictsWithCloudFrontDefaultCertificate (230.13s)
--- PASS: TestAccAWSCloudFrontDistribution_Enabled (567.94s)
--- PASS: TestAccAWSCloudFrontDistribution_S3OriginWithTags (568.65s)
--- PASS: TestAccAWSCloudFrontDistribution_OriginGroups (346.83s)
--- PASS: TestAccAWSCloudFrontDistribution_WaitForDeployment (386.97s)

--- FAIL: TestAccAWSCloudFrontDistribution_originPolicy (11.53s) -- unrelated to changes here; relates to #17342

```

File: aws/resource_aws_cloudfront_realtime_log_config.go (hunk 5 above)
Comment (ewbankkit) [reply aws/resource_aws_cloudfront_realtime_log_config.go]: In new commit (added `name` variable).

**Comment by ewbankkit** [comment]
@anGie44 Nits picked.

File: aws/resource_aws_cloudfront_realtime_log_config.go [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+	return resourceAwsCloudFrontRealtimeLogConfigRead(d, meta)
+}
+
+func resourceAwsCloudFrontRealtimeLogConfigRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	logConfig, err := finder.RealtimeLogConfigByARN(conn, d.Id())
+
+	if tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig) {
```
Comment (anGie44) [inline aws/resource_aws_cloudfront_realtime_log_config.go:118]:
```suggestion
	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig) {
```

File: aws/resource_aws_cloudfront_realtime_log_config.go [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+	_, err := conn.DeleteRealtimeLogConfig(&cloudfront.DeleteRealtimeLogConfigInput{
+		ARN: aws.String(d.Id()),
+	})
+
+	if tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchRealtimeLogConfig) {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error deleting Route53 Resolver Query Log Config (%s): %w", d.Id(), err)
```
Comment (anGie44) [inline aws/resource_aws_cloudfront_realtime_log_config.go:183]:
```suggestion
		return fmt.Errorf("error deleting Cloudfront Real-time Log Config (%s): %w", d.Id(), err)
```

**Review by anGie44 — COMMENTED** [review]
thanks @ewbankkit ! Last 2 I promise  🤣

**Pull request merged**